File: leetcode/selfDevidingNumber.go
func selfDividingNumbers(left int, right int) []int {
    var results []int
    for i := left; i <= right; i++ {
        stri := strconv.Itoa(i)
        digit_array := make([]int, len(stri))
        for j, ch := range stri {
            digit_array[j], _ = strconv.Atoi(string(ch))
        }
        accepted := true
        for _, value := range digit_array {
            if value == 0 {
                accepted = false
                break
            }
            if i % value != 0 {
                accepted = false
                break
            }
        }
        if accepted {
            results = append(results, i)
        }
    }
    return results
}

File: leetcode/selfDevidingNumber.php
<?php
class Solution {

    /**
     * @param Integer $left
     * @param Integer $right
     * @return Integer[]
     */
    function selfDividingNumbers($left, $right) {
        
        $results = [];
        for ($i = $left; $i <= $right; $i++) {
            $stri = (string)$i;
            $digit_array = array_map('intval', str_split($stri));
            $accepted = false;
            foreach($digit_array as &$value) {

                if (fmod($i,$value) == 0) {
                    $accepted = true;
                } else {
                    $accepted = false;
                    break;
                }
            }
            if ($accepted) {
                array_push($results,$i);
            }
        }
        return $results;
    }
} 
?>